### PR TITLE
add berlin tx support

### LINKED ISSUE #3963: Support for Berlin hard-fork
With Berlin hard-fork [announced](https://blog.ethereum.org/2021/03/08/ethereum-berlin-upgrade-announcement/) to be released at 14-Apr, it would be nice if the web3 would support sending (and decoding) [EIP-2918](https://eips.ethereum.org/EIPS/eip-2718)/[EIP-2930](https://eips.ethereum.org/EIPS/eip-2930) transactions.

### PR DESCRIPTION
**Pull request by GregTheGreek**
## Description

Adds berlin tx support.

Notes about tests:
Since the new transaction type changes how the transactions are RLP'd some of the messages hashes needed to be updated. Only test cases with `error: false` (or omitted completely) will run successfully in the first place, and thus tests with `error: true` did not need to be updated.

closes #3963
closes #4012 (supersedes) 
## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [x] I ran `npm run test:unit` with success.
- [x] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [x] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [x] I have tested my code on the live network.
- [x] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

**Review by luu-alex — DISMISSED** [review]

**Review by Spacesai1or — CHANGES_REQUESTED** [review]
The `dist/` files were updated, but should only be updated upon release, otherwise, it looks good!

File: test/eth.accounts.signTransaction.js [hunk 1]
```diff
@@ -58,10 +75,15 @@ var tests = [
             r: "0x22f17b38af35286ffbb0c6376c86ec91c20ecbad93f84913a0cc15e7580cd9",
             s: "0x83d6e12e82e3544cb4439964d5087da78f74cefeec9a450b16ae179fd8fe20",
             // signature from eth_signTransaction
-            rawTransaction: "0xf85d8080827c6d94f0109fc8df283027b6285cc889f5aa624eac1f558080269f22f17b38af35286ffbb0c6376c86ec91c20ecbad93f84913a0cc15e7580cd99f83d6e12e82e3544cb4439964d5087da78f74cefeec9a450b16ae179fd8fe20",
             oldSignature: "0xf85d8080827c6d94f0109fc8df283027b6285cc889f5aa624eac1f558080269f22f17b38af35286ffbb0c6376c86ec91c20ecbad93f84913a0cc15e7580cd99f83d6e12e82e3544cb4439964d5087da78f74cefeec9a450b16ae179fd8fe20",
-            transactionHash: "0xb0c5e2c6b29eeb0b9c1d63eaa8b0f93c02ead18ae01cb7fc795b0612d3e9d55a",
-            messageHash: "0x21975b15072795e610d2937abcf15ed4aecd0650b8a62204274bef3b57a8501a"
+            // for gasPrice = 0
```
Comment (Spacesai1or) [inline test/eth.accounts.signTransaction.js]: Should this be removed?

File: test/eth.accounts.signTransaction.js [hunk 2]
```diff
@@ -106,10 +128,15 @@ var tests = [
         r: "0x22f17b38af35286ffbb0c6376c86ec91c20ecbad93f84913a0cc15e7580cd9",
         s: "0x83d6e12e82e3544cb4439964d5087da78f74cefeec9a450b16ae179fd8fe20",
         // signature from eth_signTransaction
-        rawTransaction: "0xf85d8080827c6d94f0109fc8df283027b6285cc889f5aa624eac1f558080269f22f17b38af35286ffbb0c6376c86ec91c20ecbad93f84913a0cc15e7580cd99f83d6e12e82e3544cb4439964d5087da78f74cefeec9a450b16ae179fd8fe20",
         oldSignature: "0xf85d8080827c6d94f0109fc8df283027b6285cc889f5aa624eac1f558080269f22f17b38af35286ffbb0c6376c86ec91c20ecbad93f84913a0cc15e7580cd99f83d6e12e82e3544cb4439964d5087da78f74cefeec9a450b16ae179fd8fe20",
-        transactionHash: "0xb0c5e2c6b29eeb0b9c1d63eaa8b0f93c02ead18ae01cb7fc795b0612d3e9d55a",
-        messageHash: "0x21975b15072795e610d2937abcf15ed4aecd0650b8a62204274bef3b57a8501a"
+        // for gasPrice = 0
```
Comment (Spacesai1or) [inline test/eth.accounts.signTransaction.js]: Should this be removed?

**Comment by GregTheGreek** [comment]
Currently blocked by https://github.com/ethereumjs/ethereumjs-monorepo/pull/1292 but local testing shows it solves our problems :confetti_ball:

**Pull request closed without merging**